src/Server/RestRouterHandlers/TableRestRouterHandler.h

```diff
@@ -44,7 +47,7 @@ class TableRestRouterHandler : public RestRouterHandler
         const Poco::JSON::Object::Ptr & payload, const String & time_column, const String & default_order_by_granularity) const = 0;
 
 private:
-    bool tableExists(const String & database, const String & table) const;
+    CatalogService & catalog;
```

Comment:
this is not actually quite good. Let's use the `instance()` function to get catalog